src/index.ts

```diff
@@ -63,21 +61,22 @@ export const createError = (name:string, config: ErrorConfig) => {
   return e;
 };
 
-export const formatError = (error, returnNull = false) => {
-  const originalError = error ? error.originalError || error : null;
+export const formatError = (error = <any>{}, returnNull = false) => {
+  const returnVal = returnNull ? null : error;
 
-  if (!originalError) return returnNull ? null : error;
+  let originalError = error.originalError;
+  if (!originalError) return returnVal;
 
   const { name } = originalError;
+  if (!name || !isApolloError(originalError)) return returnVal;
 
-  if (!name || !isInstance(originalError)) return returnNull ? null : error;
-
-  const { time_thrown, message, data, _showLocations } = originalError;
-
-  if (_showLocations) {
+  if (originalError._showLocations) {
     const { locations, path } = error;
-    originalError.locations = locations;
-    originalError.path = path;
+    originalError = {
```

Comment:
is this necessary?

Reply:
originalError will not be the original error if it is mutated, how do you get the original error otherwise.

Reply:
Well, I believe in the original we are just mutating the reference and decorating it with new data rather than shallow cloning it like you are doing here to create a new reference.  I guess that's what I am asking: Is it necessary to create a new shallow reference copy and have it immutable in this case?